connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -356,14 +433,56 @@ private TaskStatus parseTaskStatus(ConnectorTaskId taskId, byte[] data) {
         }
     }
 
+    private TopicStatus parseTopicStatus(byte[] data) {
```

Comment:
Would be nice to add a unit test for this method, which would require this to be protected for visibility to unit tests.